src/test_application/flight.cpp

```diff
@@ -0,0 +1,167 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2020 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "flight.h"
+
+#include "DebugObject.h"
+
+#include <osp/Active/ActiveScene.h>
+#include <osp/Active/SysDebugRender.h>
+#include <osp/Active/SysVehicle.h>
+#include <osp/Active/SysForceFields.h>
+#include <osp/Active/SysAreaAssociate.h>
+
+#include <adera/Machines/UserControl.h>
+#include <adera/Machines/Rocket.h>
+
+#include <planet-a/Active/SysPlanetA.h>
+#include <planet-a/Satellites/SatPlanet.h>
+
+using osp::Vector2;
+
+using osp::Vector3;
+using osp::Quaternion;
+using osp::Matrix3;
+using osp::Matrix4;
+
+// for the 0xrrggbb_rgbf and angle literals
+using namespace Magnum::Math::Literals;
+
+using osp::universe::Universe;
+using osp::universe::Satellite;
+using osp::universe::SatActiveArea;
+
+using osp::universe::UCompActiveArea;
+
+using osp::active::ActiveEnt;
+using osp::active::ACompTransform;
+using osp::active::ACompCamera;
+using osp::active::ACompFloatingOrigin;
+
+using adera::active::machines::SysMachineUserControl;
+using adera::active::machines::SysMachineRocket;
+
+using planeta::universe::SatPlanet;
+
+
+void test_flight(std::unique_ptr<OSPMagnum>& magnumApp,
+                 osp::OSPApplication& ospApp, OSPMagnum::Arguments args)
+{
+
+    // Get needed variables
+    Universe &uni = ospApp.get_universe();
+    SatActiveArea *satAA = static_cast<osp::universe::SatActiveArea*>(
```

Comment:
You should assert that a dynamic cast results in a non-nullptr value. For an extra sanity check.

Reply:
This line actually turns out to be a mistake that happens to compile and work. The entire line is:
```cpp
SatActiveArea *satAA = static_cast<osp::universe::SatActiveArea*>(
            uni.sat_type_find("Vehicle")->second.get());
```
It's searching for "Vehicle" even though it's expecting a SatActiveArea. It just happens that none of the subsequent functions don't access any member variables. The Sat* classes already has a get_name function, which means that specifying the name in this function isn't even needed, avoiding these mistakes entirely.

RTTI is avoided and isn't used anywhere. Not even any of the dependencies use it. Perhaps just an assert would do? I might also put the casting into a template and use the get_name function, to be able to do this instead:

```cpp
auto &satAA = uni.sat_type_find<osp::universe::SatActiveArea>()
```